### DISCUSSION TITLE
Large difference in steps per second with 'spring' vs 'generalized'

### DISCUSSION THREAD

**Discussion by gijskoning** [google/brax#315] [General]
Is it expected that Ant with the generalized pipeline runs 40 times slower compared to the Spring pipeline? (Generalized sps: ~2100, spring: ~85000)

I am running the regular Brax v2 training loop with GPU Nvidia T1000 and actually wanted to train on Halfcheetah but found out that Generalized with any contact pairs runs very slowly. And I din't expect this big of a difference

**Comment by btaba** [comment]
Hi @gijskoning , thanks for the question! We may tune generalized solver params a bit for each env for speed,  but it's expected that  generalized  is the slower backend

**Comment by erikfrey** [comment]
After tuning is complete we expect generalized to be ~5x slower than spring.
